test/node/node.test.js

```diff
@@ -2,78 +2,37 @@
-                expectedNodeArgs: ['--name1=value1', '--name2="value2"', '--name3', 'value3', '-k', 'v'],
-            },
-        ].map(({ rawArgs, expectedNodeArgs, expectedCliArgs }) => {
-            const { nodeArgs, cliArgs } = parseNodeArgs(rawArgs);
-            expect(nodeArgs).toEqual(expectedNodeArgs);
-            expect(cliArgs).toEqual(expectedCliArgs);
-        });
-    });
-
-    it('is able to pass the options flags to node js', (done) => {
-        const { stdout } = run(
-            __dirname,
-            [
-                '--node-args',
-                `--require=${resolve(__dirname, 'bootstrap.js')}`,
-                '--node-args',
-                `-r ${resolve(__dirname, 'bootstrap2.js')}`,
-                '--output',
-                './bin/[name].bundle.js',
-            ],
-            false,
-        );
+    it('is able to pass the options flags to node js', async (done) => {
+        const { stdout, stderr } = await run(__dirname, ['--output', './bin/[name].bundle.js'], false, [
+            `--require=${resolve(__dirname, 'bootstrap.js')}`,
+            `--require=${resolve(__dirname, 'bootstrap2.js')}`,
+        ]);
         expect(stdout).toContain('---from bootstrap.js---');
         expect(stdout).toContain('---from bootstrap2.js---');
+        expect(stderr).toBeFalsy();
         stat(resolve(__dirname, './bin/main.bundle.js'), (err, stats) => {
             expect(err).toBe(null);
             expect(stats.isFile()).toBe(true);
             done();
         });
     });
 
-    it('throws an error on supplying unknown flags', () => {
-        const { stderr } = run(__dirname, ['--node-args', '--unknown']);
+    it('throws an error on supplying unknown flags', async () => {
+        const { stderr } = await run(__dirname, [], false, ['--unknown']);
         expect(stderr).toContain('bad option');
     });
 
-    it('throws an error if no values were supplied with --max-old-space-size', () => {
-        const { stderr, stdout } = run(__dirname, ['--node-args', '--max-old-space-size']);
+    it('throws an error if no values were supplied with --max-old-space-size', async () => {
+        const { stderr, stdout } = await run(__dirname, [], false, ['--max-old-space-size']);
         expect(stderr).toContain('missing value for flag --max-old-space-size');
         expect(stdout).toBeFalsy();
     });
 
-    it('throws an error if an illegal value was supplied with --max-old-space-size', () => {
-        const { stderr, stdout } = run(__dirname, ['--node-args', '--max-old-space-size=1024a']);
-        expect(stderr).toContain('illegal value for flag --max-old-space-size');
+    it('throws an error if an illegal value was supplied with --max-old-space-size', async () => {
+        const { stderr, stdout } = await run(__dirname, [], true, ['--max_old_space_size=1024a']);
+        expect(stderr).toContain('Error: illegal value for flag --max_old_space_size=1024a of type size_t');
```

Comment:
Can we add tests for `NODE_OPTIONS` env variable?

Reply:
Using inbuilt env to pass `NODE_OPTIONS` from execa throws different error messages since it sets on process.env so I'm guessing it's past the validation stage. https://github.com/sindresorhus/execa#env hence set NODE_OPTIONS via the `nodeOption` param

Directly passing NODE_OPTIONS doesn't seem to work correctly 
![image](https://user-images.githubusercontent.com/21009455/90168202-636d8980-ddba-11ea-9765-459f23526d2a.png)

Reply:
Different err when passing it via execa env param, I'm guessing it does some validation for the passed NODE_OPTIONS before setting to env, but here we're setting in env directly bypassing the validation hence different err
![image](https://user-images.githubusercontent.com/21009455/90168372-a2034400-ddba-11ea-951e-5819a7620d45.png)

Reply:
interesting, need investigate, can you add `TODO` with `NODE_OPTIONS` to add test on this stuff and test it manually?

Reply:
Yep added comment 👍 

Working fine otherwise 
![image](https://user-images.githubusercontent.com/21009455/90170615-ef34e500-ddbd-11ea-83b1-c47627fb8e98.png)